addons/stock/models/stock_move.py

```diff
@@ -602,13 +602,53 @@ def _propagate_date_log_note(self, move_orig):
             odoobot_id = self.env['ir.model.data'].xmlid_to_res_id("base.partner_root")
             doc.message_post(body=msg, author_id=odoobot_id, subject=msg_subject)
 
+    def _prefill_return_move_lines(self):
+        """ Ease encoding of returns with tracked products by prefilling move lines
+        """
+        if self.origin_returned_move_id and float_is_zero(self.quantity_done, precision_rounding=self.product_uom.rounding):
+            move_lines_commands = []
+            origin_quantity = self.origin_returned_move_id.product_uom._compute_quantity(self.origin_returned_move_id.product_uom_qty, self.product_uom, round=False)
+            if self.product_id.tracking == 'lot':
+                if float_compare(self.product_qty, origin_quantity, precision_rounding=self.product_uom.rounding) == 0:
+                    for move_line in self.origin_returned_move_id.move_line_ids:
+                        move_line_vals = self._prepare_move_line_vals(quantity=0)
+                        move_line_vals['qty_done'] = move_line.product_uom_id._compute_quantity(move_line.qty_done, self.product_uom)
+                        move_line_vals['product_uom_id'] = self.product_uom.id
+                        move_line_vals['lot_id'] = move_line.lot_id.id
+                        move_lines_commands.append((0, 0, move_line_vals))
+                elif len(self.origin_returned_move_id.move_line_ids.lot_id) == 1:
+                    move_line_vals = self._prepare_move_line_vals(quantity=0)
```

Comment:
you should remove `quantity=0` at each call of `_prepare_move_line_vals`, it is the same that don't put it and then it is useless. Or I miss something ?

Reply:
Some have, some don't : I will remove.
About move_line_vals['qty_done'] = 0, this is to point out the fact that we can't prefill quantity done, only lot id.
Should I remove, comment or make a note ?

Reply:
IMO, you should remove it with maybe a small comment instead, `# can't prefill quantity done bacause ...`.